src/core/build-configuration.ts

```diff
@@ -0,0 +1,29 @@
+
+/** @ignore not in use for the moment */
+export class BuildConfiguration {
+    private static instance: BuildConfiguration;
+    private devMode: boolean = true;
+    private runModeLocked: boolean = false;
+
+    private constructor() { }
+
+    public enableProdMode(): void {
+        if (this.runModeLocked) {
+            throw new Error('Cannot enable prod mode after cli setup.');
+        }
+        this.runModeLocked = true;
+        this.devMode = false;
+    }
+
+    public isDevMode(): boolean {
+        this.runModeLocked = true;
+        return this.devMode;
+    }
+
+    public static getInstance(): BuildConfiguration {
```

Comment:
Why did you do it like this? Just keep it as functions (that's how Ng does it). That way, it's always referring to the global variable. (TS has gotten to you 😄  not everything needs to be in a class)

Reply:
I can't find nothing against so I think you're right ^^